lib/util.ts

```diff
@@ -41,18 +41,22 @@ export function runFilenameOrFn_(
     if (filenameOrFn &&
         !(typeof filenameOrFn === 'string' ||
           typeof filenameOrFn === 'function')) {
-      throw 'filenameOrFn must be a string or function';
+      throw new Error('filenameOrFn must be a string or function');
     }
 
     if (typeof filenameOrFn === 'string') {
       filenameOrFn = require(resolve(configDir, filenameOrFn));
     }
     if (typeof filenameOrFn === 'function') {
       let results = when(filenameOrFn.apply(null, args), null, (err) => {
-        if (err.stack) {
+        if (typeof err === 'string') {
```

Comment:
Should we check `if (!err.stac)` instead of `typeof`? In case we get some crazy error object without a stack or something?